src/oasis-indexer/api.ts

```diff
@@ -17,42 +16,48 @@ function fromBaseUnits(valueInBaseUnits: string, decimals: number): string {
   return value.toFixed()
 }
 
-export const useGetEmeraldTransactions = (
-  params?: generated.GetEmeraldTransactionsParams,
-  options?: { query: UseQueryOptions<AxiosResponse<generated.RuntimeTransactionList>> },
-) => {
+function arrayify<T>(arrayOrItem: null | undefined | T | T[]): T[] {
+  if (arrayOrItem == null) return []
+  if (!Array.isArray(arrayOrItem)) return [arrayOrItem]
+  return arrayOrItem
+}
+
+export const useGetEmeraldTransactions: typeof generated.useGetEmeraldTransactions = (params?, options?) => {
   const [searchParams] = useSearchParams()
   const offsetSearchQuery = searchParams.get('offset')
   const offset = (offsetSearchQuery && parseInt(offsetSearchQuery, 10)) || 0
 
-  const result = generated.useGetEmeraldTransactions({ ...params, offset }, options)
-  if (result.data) {
-    return {
-      ...result,
-      data: {
-        ...result.data,
-        data: {
-          ...result.data.data,
-          transactions: result.data.data.transactions?.map(tx => {
+  const result = generated.useGetEmeraldTransactions(
+    { ...params, offset },
+    {
+      ...options,
+      axios: {
+        ...options?.axios,
+        transformResponse: [
```

Comment:
Converted parsing into axios transform instead of parsing after query returns:
- i wanted structural sharing to have the parsed data
- i tried using react-query's select function, but it makes a bunch of weirdness in structuralSharing